jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java

```diff
@@ -1777,40 +1667,152 @@ public Object get(String elementName, boolean mandatory) throws Exception
+                Annotation[][] parameterAnnotations = executable.getParameterAnnotations();
+                if (parameterAnnotations == null || parameterAnnotations.length == 0)
+                    return null;
+
+                // Find the position of all named parameters from the executable
+                Map<String, Integer> position = new HashMap<>();
+                int p = 0;
+                for (Annotation[] paramAnnotation : parameterAnnotations)
                 {
-                    Object o = _node.get(i);
-                    if (!(o instanceof XmlParser.Node))
-                        continue;
-                    XmlParser.Node n = (XmlParser.Node)o;
+                    Integer pos = p++;
+                    Arrays.stream(paramAnnotation)
+                        .filter(Name.class::isInstance)
+                        .map(Name.class::cast)
+                        .findFirst().ifPresent(n -> position.put(n.value(), pos));
+                }
 
-                    if (elementName.equals(n.getTag()))
+                List<Object> arguments = new ArrayList<>(_arguments);
+                List<String> names = new ArrayList<>(_names);
+                // Map the actual arguments to the names
+                for (p = 0; p < count; p++)
+                {
+                    String name = names.get(p);
+                    if (name != null)
                     {
-                        if (attr != null)
-                            throw new IllegalStateException("Cannot have attr '" + attrName + "' and element '" + elementName + "'");
-
-                        values.add(n);
+                        Integer pos = position.get(name);
+                        if (pos == null)
+                            return null;
+                        if (pos != p)
+                        {
+                            // adjust position of parameter
+                            arguments.add(pos, arguments.remove(p));
+                            names.add(pos, names.remove(p));
+                            p = Math.min(p, pos);
+                        }
                     }
                 }
+                return arguments.toArray(new Object[0]);
+            }
+        }
+    }
 
-                return values;
+    private static List<XmlParser.Node> getNodes(XmlParser.Node node, String elementName)
+    {
+        String attrName = StringUtil.asciiToLowerCase(elementName);
+        final List<XmlParser.Node> values = new ArrayList<>();
+
+        String attr = node.getAttribute(attrName);
+        if (attr != null)
+        {
+            for (String a : StringUtil.csvSplit(null, attr, 0, attr.length()))
+            {
+                // create a fake node
+                XmlParser.Node n = new XmlParser.Node(null, elementName, null);
+                n.add(a);
+                values.add(n);
             }
         }
+
+        for (int i = 0; i < node.size(); i++)
+        {
+            Object o = node.get(i);
```

Comment:
This can be `for (Object o : node)`

Reply:
There are many such for loops in the class, so I tried to convert all.... but broke something... obviously my virus infected mind is not up to such complex programming :)   Let's leave as is for now!